src/etaoin/api.clj

```diff
@@ -3546,9 +3547,29 @@
                          (drv/set-capabilities (:capabilities defaults-global))
                          (drv/set-capabilities (get-in defaults [type :capabilities]))
                          (drv/set-capabilities capabilities)))
-        caps          (:capabilities driver)
-        session       (create-session driver caps)]
-    (assoc driver :session session)))
+        caps          (:capabilities driver)]
+    (loop [n 1]
+      ;; Wait for driver to be ready before creating the first session.
+      (wait-predicate
+       (fn [] (-> (get-status driver) :ready))
+       {:timeout 30
+        :interval 0.100
+        :message "Timeout waiting for WebDriver to become ready after creation."})
+      (let [driver (assoc driver :session (create-session driver caps))
+            good-session (try (get-window-handle driver)
```

Comment:
OK, will do. I was thinking the same thing after I submitted the PR, but didn't want to rock the boat with more changes. Yes, I think it's OK for all browsers as they should act correctly. If they don't, then it might help them, too, but in any case shouldn't cause more problems with other browsers than they would otherwise encounter.

Reply:
Done.